### PR TITLE
Repo server rbac

### PR DESCRIPTION
**Pull request by fiftoine**
Checklist:

* [V] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [V] Any new values are backwards compatible and/or have sensible default.
* [V] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [V] I have signed the CLA and the build is green.
* [V] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by fiftoine** [comment]
Solving issue https://github.com/argoproj/argo-helm/issues/154

**Comment by seanson** [comment]
Looks good, just needs a chart version bump!

**Comment by fiftoine** [comment]
> Looks good, just needs a chart version bump!

Bumped ;-)

File: charts/argo-cd/templates/argocd-repo-server/deployment.yaml [hunk 1]
```diff
@@ -104,6 +104,9 @@ spec:
     {{- if .Values.repoServer.affinity }}
       affinity:
 {{- toYaml .Values.repoServer.affinity | nindent 8 }}
+    {{- end }}
+    {{- if .Values.repoServer.serviceAccount.create }}
```
Comment (seanson) [inline charts/argo-cd/templates/argocd-repo-server/deployment.yaml]: Could you please remove this conditional so we can support the three use cases:
- create: false, service account name unset: template defaults to serviceAccountName: "default"
- create: false, service account name set: template uses specific service account name (common use case for externally created service accounts)
- create: true: service account name set: new service account created

File: charts/argo-cd/templates/argocd-repo-server/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.repoServer.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "argo-cd.repoServerServiceAccountName" . }}
+  labels:
```
Comment (seanson) [inline charts/argo-cd/templates/argocd-repo-server/serviceaccount.yaml]: Could you please add the option to set annotations on the service account. This is required for things like workload identity on Google Cloud and AWS and would be very helpful for any reposerver that needs to pull info from S3 or google storage.

File: charts/argo-cd/values.yaml [hunk 3]
```diff
@@ -480,6 +480,23 @@ repoServer:
     #   namespace: monitoring
     #   additionalLabels: {}
 
+  ## Repo server service account
+  ## If create is set to true, make sure to uncomment the name and update the rbac section below
+  serviceAccount:
+    create: false
+  #  name: argocd-repo-server
```
Comment (seanson) [inline charts/argo-cd/values.yaml]: As per above, needs a new field for service account annotations.

**Comment by fiftoine** [comment]
> Awesome work @fiftoine! I did some testing on my side and there are a few small changes we'd need to make in order to cover some common reposerver service account use cases.

All done ;-)

**Pull request merged**